### PR TITLE
refactor: alert component tweaks and update icons

### PR DESCRIPTION
**Pull request by Highjhacker**
## Checklist

- [ ] Documentation _(if necessary)_
- [ ] Tests _(if necessary)_
- [x] Ready to be merged

### CODE REVIEW

**Review by ItsANameToo — CHANGES_REQUESTED** [review]
please check the `alert-simple` classes too, as the styleguide has slighty different styling for those

**Comment by ItsANameToo** [comment]
🤔 maybe we don't even use alert-simple anymore, I can't find a reference to it at least. Can you check and remove if obsolete @Highjhacker ?

**Comment by ItsANameToo** [comment]
ah looks like we use it for deployer @Highjhacker

**Comment by Highjhacker** [comment]
@ItsANameToo I'll have a look, looked a bit around on MSQ and tried the alert-simple myself but yeah, no point of reference design wise. Will check on the styleguide and correct :vinigotit:

File: resources/assets/icons/exclamation-mark.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg version="1.1" id="Слой_1" xmlns="http://www.w3.org/2000/svg" x="0" y="0" viewBox="0 0 2.8 12.2" xml:space="preserve"><style>.exclamation-mark-st0{fill:none;stroke:currentColor;stroke-width:2;stroke-linecap:round;stroke-linejoin:round}</style><g id="warning_2" transform="translate(-649.6 -5965)"><g id="Group_586"><g id="remove-circle-bold"><path id="Shape_2591" class="exclamation-mark-st0" d="M651 5966v6"/><path id="Shape_2592" class="exclamation-mark-st0" d="M651 5975.4c-.2 0-.4.2-.4.4s.2.4.4.4.4-.2.4-.4-.2-.4-.4-.4"/></g></g></g></svg>
```
Comment (ItsANameToo) [inline resources/assets/icons/exclamation-mark.svg:1]: image
 sus

File: resources/assets/icons/success.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 11.9 10.3"><path d="M1.397 6.304l1.9 2.6c.2.3.6.5 1 .3.1 0 .1 0 .1-.1h0c.1 0 .1-.1.2-.2l5.9-7.5" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/></svg>
```
Comment (ItsANameToo) [inline resources/assets/icons/success.svg:1]: we already have a `checkmark` and `checkmark-thin` and this is just another variation of it. Check with oleg if it should replace the `checkmark-thin` one as that one is similar in shape

File: resources/views/alert.blade.php [hunk 3]
```diff
@@ -3,17 +3,18 @@
     'messageClass' => 'text-sm',
     'type' => 'info',
     'large' => false,
+    'rounded' => false,
```
Comment (ItsANameToo) [inline resources/views/alert.blade.php:6]: is this in the designs that we need a rounded icon version?

File: resources/assets/icons/danger.svg [hunk 4]
```diff
@@ -0,0 +1 @@
+<svg version="1.1" id="Слой_1" xmlns="http://www.w3.org/2000/svg" x="0" y="0" viewBox="0 0 11.8 11.8" xml:space="preserve"><style>.danger-st0{fill:none;stroke:currentColor;stroke-width:2;stroke-linecap:round;stroke-linejoin:round}</style><g id="danger_2" transform="translate(-453.086 -4389.086)"><g id="remove-circle"><path id="Shape_1779" class="danger-st0" d="M454.5 4399.5l9-9"/><path id="Shape_1780" class="danger-st0" d="M463.5 4399.5l-9-9"/></g></g></svg>
```
Comment (ItsANameToo) [inline resources/assets/icons/danger.svg:1]: we already have `cross` and `close` that are similar, should it replace one of those?

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

File: resources/views/alert.blade.php (hunk 3 above)
Comment (Highjhacker) [reply resources/views/alert.blade.php:6]: Went through the designs and nope, not for MSQ at least, worth checking elsewhere or I just trash the whole rounded ? 🤔

Comment (ItsANameToo) [reply resources/views/alert.blade.php:6]: ask oleg

File: resources/assets/icons/danger.svg (hunk 4 above)
Comment (Highjhacker) [reply resources/assets/icons/danger.svg:1]: Just need to delete the `cross` one and we keep the rest according to Oleg.

File: resources/views/alert.blade.php (hunk 3 above)
Comment (Highjhacker) [reply resources/views/alert.blade.php:6]: Yeah we ditch the round icons for alerts, so removing this.

File: src/helpers.php [hunk 5]
```diff
@@ -21,3 +21,21 @@ function formatReadTime(int $minutes): string
...
 }
+
+if (! function_exists('alertIcon')) {
+    function alertIcon(string $type): string
+    {
+        if (in_array($type, ['success', 'error', 'danger', 'hint', 'warning', 'info'])) {
+            return [
+                'success' => 'checkmark-smooth',
+                'error' => 'danger',
+                'danger' => 'danger',
```
Comment (ItsANameToo) [inline src/helpers.php:32]: `danger` icon no longer exists

**Review by ItsANameToo — CHANGES_REQUESTED** [review]
the old alert icons are still present too, those need to be removed

File: src/UserInterfaceServiceProvider.php [hunk 6]
```diff
@@ -21,12 +21,12 @@ class UserInterfaceServiceProvider extends ServiceProvider
...
-            'warning' => 'alert-warning',
-            'danger'  => 'alert-danger',
-            'error'   => 'alert-error',
-            'hint'    => 'alert-hint',
+            'info'    => 'info',
+            'success' => 'checkmark-smooth',
+            'warning' => 'exclamation-mark',
+            'danger'  => 'cross',
+            'error'   => 'cross',
+            'hint'    => 'question-mark',
```
Comment (ItsANameToo) [inline src/UserInterfaceServiceProvider.php:29]: doesn't this have to do with the classes applied to the flash messages?

File: src/Components/FlashMessage.php [hunk 7]
```diff
@@ -28,6 +28,6 @@ public function handleIncomingMessage(array $message): void
 
     public function alertType(): string
     {
-        return 'alert-'.$this->type;
+        return $this->type;
```
Comment (ItsANameToo) [inline src/Components/FlashMessage.php:31]: 🤔

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

File: src/UserInterfaceServiceProvider.php (hunk 6 above)
Comment (Highjhacker) [reply src/UserInterfaceServiceProvider.php:29]: It does, had to check the Spatie doc, thought it'd be about the icons, my bad

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**